extensions/amp-image-slider/0.0/amp-image-slider.js

```diff
@@ -0,0 +1,566 @@
+      this.element.removeEventListener('click', this.handleClickImage);
+      this.element.removeEventListener('touchend', this.handleTapImage);
+      dev().assertElement(this.barButton_)
+          .removeEventListener('mousedown', this.dragStart);
+      dev().assertElement(this.barButton_)
+          .removeEventListener('touchstart', this.touchStart);
+      // remove pointer related events below
+      this.dragEnd(null);
+      this.touchEnd(null);
+    }
+
+    this.isEventRegistered_ = false;
+  }
+
+  /**
+   * Handle hover event
+   * @param {Event} e
+   */
+  handleHover(e) {
+    // This offsetWidth may change if user resize window
+    // Thus not cached
+    // const {left, width} = this.getLayoutBox();
+    const {left, width} = this.container_./*OK*/getBoundingClientRect();
+    const leftPercentage = (e.pageX - left) / width;
+    this.updatePositions(leftPercentage);
+  }
+
+  /**
+   * Handle click on the image
+   * @param {Event} e
+   */
+  handleClickImage(e) {
+    // const {left, width} = this.getLayoutBox();
+    const {left, width} = this.container_./*OK*/getBoundingClientRect();
+    const leftPercentage = (e.pageX - left) / width;
+    this.animateUpdatePositions(leftPercentage);
+  }
+
+  /**
+   * Handle tap on the image
+   * @param {Event} e
+   */
+  handleTapImage(e) {
+    // const {left, width} = this.getLayoutBox();
+    const {left, width} = this.container_./*OK*/getBoundingClientRect();
+    if (e.touches.length > 0) {
+      const leftPercentage = (e.touches[0].pageX - left) / width;
+      this.animateUpdatePositions(leftPercentage);
+    }
+  }
+
+  /**
+   * Handle hinding the hint
+   * @param {Event} unusedEvent
+   */
+  handleHideHint(unusedEvent) {
+    if (!this.isBarHintHidden_ && this.barHint_) {
+      this.isBarHintHidden_ = true;
+      this.barHint_.classList.add('i-amphtml-image-slider-hint-hidden');
+    }
+  }
+
+  /**
+   * Animated, update element positions based on percentage
+   * @param {number} toPercentage
+   */
+  animateUpdatePositions(toPercentage) {
+    const {left: containerLeft, width: containerWidth}
+    //    = this.getLayoutBox();
+        = this.container_./*OK*/getBoundingClientRect();
+    const {left: barLeft} = this.bar_./*OK*/getBoundingClientRect();
+
+    const fromPercentage = (barLeft - containerLeft) / containerWidth;
+    const interpolate = numeric(fromPercentage, toPercentage);
+    const curve = bezierCurve(0.4, 0, 0.2, 1); // fast-out-slow-in
+    const duration = 200;
+    // Single animation ensure elements have props updated at same pace
+    // Multiple animations would be scheduled at different raf
+    return Animation.animate(this.element, pos => {
+      this.updatePositions(interpolate(pos));
+    }, duration, curve).thenAlways();
+  }
+
+  /**
+   * Add listeners on drag start
+   * @param {Event} e
+   */
+  dragStart(e) {
+    e.preventDefault();
+    e.stopPropagation();
+
+    this.win.addEventListener('mousemove', this.dragMove);
```

Comment:
Can you remind me why the `listen` function from `event-helper.js` doesn't work here?